File: app/src/main/java/info/juanmendez/aademo/MainActivity.java
package info.juanmendez.aademo;

import android.app.Activity;
import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.widget.TextView;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.ViewById;

@EActivity(R.layout.activity_main)
public class MainActivity extends ActionBarActivity {

    @ViewById TextView myText;

    @AfterViews
    void sayHi(){
        myText.setText("hello AndroidAnnotations!");
    }
}

File: app/src/main/java/info/juanmendez/aademo/MainActivity.kt
package info.juanmendez.aademo

import android.support.v7.app.AppCompatActivity
import info.juanmendez.aademo.databinding.ActivityMainBinding
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.BindingObject
import org.androidannotations.annotations.DataBound
import org.androidannotations.annotations.EActivity

@DataBound
@EActivity(R.layout.activity_main)
class MainActivity:AppCompatActivity() {

    @BindingObject
    lateinit var binding:ActivityMainBinding

    @AfterViews
    fun afterViews() {
        binding.vm = VM()
    }
}